.github/workflows/ci.yml

```diff
@@ -0,0 +1,40 @@
+---
+name: CI
+
+on:
+  - pull_request
+  - push
+
+env:
+  TESTOPT: "-v"
+
+concurrency:
+  group: ${{ github.ref_name }}
+  cancel-in-progress: true
+
+jobs:
+  test:
+    name: Ruby
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby:
+          - "2.0"
+          - "2.1"
+          - "2.2"
+          - "2.3"
+          - "2.4"
+          - "2.5"
+          - "2.6"
+          - "2.7"
+          - "3.0"
+          - "3.1"
+          - "ruby-head"
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+          bundler-cache: true
```

Comment:
You'd have to use the github cache action as well for persisting the cache across runs, right?

Reply:
This is all built into the ruby/setup-ruby action. See https://github.com/ruby/setup-ruby#caching-bundle-install-automatically for the details.

Reply:
Ah, TIL! Thanks for the info @ekohl!